src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -18145,8 +18145,102 @@ def dt_strftime(self, date_format: str) -> None:
         Returns:
             New QueryCompiler containing formatted date-time values.
         """
-        ErrorMessage.not_implemented(
-            "Snowpark pandas doesn't yet support the method 'Series.dt.strftime'"
+
+        def strftime_func(column: SnowparkColumn) -> SnowparkColumn:
```

Comment:
I am wondering if it's possible to use `to_char` method with format. And translate pandas directives to snowflake directives https://docs.snowflake.com/en/sql-reference/functions-conversion#date-and-time-formats-in-conversion-functions
I believe that will be more efficient solution?

Reply:
Yes, great idea. Will try that.

Reply:
It may not actually work in the general case. The issue is that if the text aside from the directives inside `date_format` conflicts with the Snowflake directives, `to_char` will return the wrong result (e.g., if the text contained "MM"). Also, not all pandas directives are mappable to Snowflake directives.

So, I think we can later add an optimization to first check if the given `date_format` is eligible to be implemented using `to_char`. If so, we use `to_char`. Otherwise, we use the default approach (the one in this PR).

Reply:
I see, thanks!